### PR TITLE
NOTIF-484 Render DB templates with standalone Qute engine

### PR DESCRIPTION
**Pull request by gwenneg**
This PR is necessary to allow the injection of a DB template into another DB template.

Outer template:
```
Hello, {#include inner-template /}
```
Inner template:
```
World!
```
Rendering outcome:
```
Hello, World!
```

It is based on [this part](https://quarkus.io/guides/qute-reference#standalone) of the Qute doc.

ℹ️ This PR has not impact on prod. It can be safely merged.

### CODE REVIEW

**Comment by josejulio** [comment]
Nice!

Now that you are working on this, I'm wondering if i it would be possible to know the dependent templates at compile time (of the template). So that we could somehow prevent/warn from/when deleting templates that are used in other templates.

File: engine/src/main/java/com/redhat/cloud/notifications/templates/DbTemplateLocator.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+        return new TemplateLocation() {
+
+            @Override
+            public Reader read() {
+                return new StringReader(templateData);
+            }
+
+            @Override
+            public Optional<Variant> getVariant() {
```
[the remark this answers is not in the document]
Comment (gwenneg) [reply engine/src/main/java/com/redhat/cloud/notifications/templates/DbTemplateLocator.java:49]: Qute variants are used to change the locale, content type and encoding of a template.

**Comment by gwenneg** [comment]
> Now that you are working on this, I'm wondering if i it would be possible to know the dependent templates at compile time (of the template). So that we could somehow prevent/warn from/when deleting templates that are used in other templates.

This could certainly be done manually with a SQL query which would look for `{#include template_name` in the `Template#data` field in the database. I'm not sure Qute can help with that but I'll have a deeper look.

**Comment by gwenneg** [comment]
I talked with Martin Kouba (who created Qute). Right now, there isn't anything in Qute to detect templates dependencies and prevent the deletion of a template that would be included into another one.

The short-term solution consists in doing that with SQL queries as I mentioned earlier. Because there can be several layers of nested templates, the queries will have to deal with recursive detection.

Martin invited me to create a Quarkus issue and request for [TemplateNode](https://github.com/quarkusio/quarkus/blob/main/independent-projects/qute/core/src/main/java/io/quarkus/qute/TemplateNode.java) and friends to be exposed in the Qute API. That could lead to a smarter long-term solution than the SQL queries.

**Comment by gwenneg** [comment]
I'll prepare a first (and simple) version of the "already used" check.

**Pull request merged**